tools/migrator/src/migrations/consolidate-props/transform.ts

```diff
@@ -0,0 +1,55 @@
+import type { API, FileInfo } from 'jscodeshift';
+
+import {
+  consolidateJSXAttributes,
+  ConsolidateJSXAttributesOptions,
+} from '../../utils/transformations';
+
+type TransformerOptions = ConsolidateJSXAttributesOptions & {
+  componentName: string;
+};
+
+/**
+ * Example transformer function to consolidate props
+ *
+ * @param file the file to transform
+ * @param jscodeshiftOptions an object containing at least a reference to the jscodeshift library
+ * @param options an object containing options to pass to the transform function
+ * @returns Either the modified file or the original file
+ */
+export default function transformer(
+  file: FileInfo,
+  { jscodeshift: j }: API,
+  options: TransformerOptions,
+) {
+  const source = j(file.source);
+
+  const {
+    propToRemove = 'propToRemove',
+    propToUpdate = 'propToUpdate',
+    propMapping = {
+      value2: 'value3',
+    },
+    propToRemoveType = 'string',
+    componentName = 'MyComponent',
```

Comment:
If I'm understanding correctly, these `migrations` that you've defined here are base transforms that we'll use for more specific mods?

Reply:
correct!